Resources/views/CRUD/base_edit_form_macro.html.twig

```diff
@@ -1,33 +1,53 @@
 {% macro render_groups(admin, form, groups, has_tab) %}
     {% if has_tab %}<div class="row">{% endif %}
+
+    {# Row counter for groups sizing #}
+    {% set row_started = false %}
+    {% set row_col_filled = 0 %}
+
     {% for code in groups %}
         {% set form_group = admin.formgroups[code] %}
-        <div class="{{ form_group.class }}"> {# default class 'col-md-12' removed because it do not work with tabs#}
-            <div class="box box-primary">
-                <div class="box-header">
-                    <h4 class="box-title">
-                        {{ admin.trans(form_group.name, {}, form_group.translation_domain) }}
-                    </h4>
-                </div>
-                {#<div class="box{% if loop.first %} in{% endif %}" id="{{ admin.uniqid }}_{{ loop.index }}">#}
-                <div class="box-body">
-                    <div class="sonata-ba-collapsed-fields">
-                        {% if form_group.description != false %}
-                            <p>{{ form_group.description|raw }}</p>
-                        {% endif %}
 
-                        {% for field_name in form_group.fields %}
-                            {% if admin.formfielddescriptions[field_name] is defined %}
-                                {{ form_row(form[field_name])}}
+        {# Manage calculation to determinate if a new row is necessary #}
+        {% set group_class = form_group.class | default('col-md-12') %}
```

Comment:
Ok but did you keep in mind the row management? Before my PR we had some margin issue because of non respect of .row/.col bootstrap convention.

Any demo to see the result? Thanks.